jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/ColumnBatch.java

```diff
@@ -16,23 +16,46 @@
 
 package ml.dmlc.xgboost4j.java;
 
-import java.util.Iterator;
-
 /**
- * The abstracted XGBoost ColumnBatch to get cuda array interface which is used to build
- * the DMatrix on device.
- *
+ * The abstracted XGBoost ColumnBatch to get array interface from columnar data format.
+ * For example, the cuDF dataframe which employs apache arrow specificatrion.
  */
 public abstract class ColumnBatch implements AutoCloseable {
-
   /**
    * Get the cuda array interface json string for the whole ColumnBatch including
    * the must-have feature, label columns and the optional weight, base margin columns.
    *
-   * This API will be called by {@link DMatrix#DMatrix(Iterator, float, int, int)}
-   *
+   * This function is be called by native code during iteration and can be made as private
+   * method.  We keep it as public simply to silent the linter.
    */
-  public abstract String getArrayInterfaceJson();
+  public final String getArrayInterfaceJson() {
```

Comment:
The function is considered to be internal and can be replaced by a C++ implementation.  The important part is the JSON format we pass into libxgboost.so can be changed whenever needed.